OpenEdXMobile/src/main/java/org/edx/mobile/http/OfflineRequestInterceptor.java

```diff
@@ -1,52 +1,29 @@
 package org.edx.mobile.http;
 
-import android.content.Context;
-
-import org.edx.mobile.logger.Logger;
-import org.edx.mobile.util.NetworkUtil;
-
 import java.io.IOException;
 
+import okhttp3.CacheControl;
 import okhttp3.Interceptor;
 import okhttp3.Request;
 import okhttp3.Response;
 
 /**
- * okhttp does not work with offline mode by default.
- * we force okhttp to look at the local cache when in offline mode
+ * An OkHttp interceptor that ensures that any queries which
+ * couldn't connect to the server would get a response served
+ * from the local cache instead.
  */
 public class OfflineRequestInterceptor implements Interceptor {
-    protected final Logger logger = new Logger(getClass().getName());
-    private final static int MAX_STALE = 60 * 60 * 24 * 28; // tolerate 4-weeks stale
-    private Context context;
-    private int maxStaleTime;
-    public OfflineRequestInterceptor(Context context){
-        this(context, MAX_STALE);
-    }
-
-    public OfflineRequestInterceptor(Context context, int maxStaleTime){
-
-        this.context = context;
-        this.maxStaleTime = maxStaleTime;
-    }
-
     @Override
     public Response intercept(Chain chain) throws IOException {
-        final Request.Builder requestBuilder = chain.request().newBuilder();
-        if (isNetworkConnected()) {
+        final Request request = chain.request();
+        try {
             //TODO? should we specify caching here?
-//            int maxAge = 60; // read from cache for 1 minute
-//            requestBuilder.header(("Cache-Control", "public, max-age=" + maxAge);
-            requestBuilder.header("Cache-Control", "public");
-        } else {
-            requestBuilder.header("Cache-Control",
-                    "public, only-if-cached, max-stale=" + maxStaleTime);
+//            requestBuilder.cacheControl(new CacheControl.Builder()
```

Comment:
Yeah, I didn't want to change anything in commit 6aba0741a746f1fe719bf3264d93c8e15ed5798d other than the minimum changes required to make the intercepter behave identically to previous functionality, so I didn't remove these comments here. They have been removed as obsolete along with the whole interceptor when it was replaced with the `StaleIfErrorHandlingInterceptor` though, as I don't suppose loading from the cache while online is what we want to do, and even if we did we should probably use the `stale-while-revalidate` extension, or use the `If-Modified-Since` control to send a conditional request.